### ISSUE TITLE
kyma-cli panics on macOS High Sierra / Mojave if built using go 1.15

### ISSUE THREAD

**Issue by bayandin** [kyma-project/cli#557]
**Description**

If I build kyma-cli using go 1.15 on High Sierra or Mojave it panics:
```
$ kyma-darwin 
runtime: netpoll: break fd ready for -2
fatal error: runtime: netpoll: break fd ready for something unexpected

runtime stack:
runtime.throw(0x32b9bd8, 0x39)
	/usr/local/Cellar/go/1.15/libexec/src/runtime/panic.go:1116 +0x72
runtime.netpoll(0x98fcb, 0x49fa4dc2e01)
	/usr/local/Cellar/go/1.15/libexec/src/runtime/netpoll_kqueue.go:147 +0x33e
runtime.findrunnable(0xc000058000, 0x0)
	/usr/local/Cellar/go/1.15/libexec/src/runtime/proc.go:2434 +0x785
runtime.schedule()
	/usr/local/Cellar/go/1.15/libexec/src/runtime/proc.go:2669 +0x2d7
runtime.park_m(0xc000001200)
	/usr/local/Cellar/go/1.15/libexec/src/runtime/proc.go:2837 +0x9d
runtime.mcall(0x80000)
	/usr/local/Cellar/go/1.15/libexec/src/runtime/asm_amd64.s:318 +0x5b

goroutine 1 [runnable, locked to thread]:
golang.org/x/text/language.init.0()
	/Users/bayandin/Library/Caches/Homebrew/go_mod_cache/pkg/mod/golang.org/x/text@v0.3.2/language/match.go:715 +0x325

goroutine 18 [chan receive]:
k8s.io/klog.(*loggingT).flushDaemon(0x46228a0)
	/Users/bayandin/Library/Caches/Homebrew/go_mod_cache/pkg/mod/k8s.io/klog@v1.0.0/klog.go:1010 +0x8b
created by k8s.io/klog.init.0
	/Users/bayandin/Library/Caches/Homebrew/go_mod_cache/pkg/mod/k8s.io/klog@v1.0.0/klog.go:411 +0xd8

goroutine 6 [select]:
go.opencensus.io/stats/view.(*worker).start(0xc0004d8aa0)
	/Users/bayandin/Library/Caches/Homebrew/go_mod_cache/pkg/mod/go.opencensus.io@v0.22.0/stats/view/worker.go:154 +0x105
created by go.opencensus.io/stats/view.init.0
	/Users/bayandin/Library/Caches/Homebrew/go_mod_cache/pkg/mod/go.opencensus.io@v0.22.0/stats/view/worker.go:32 +0x57
```

```
$ go version
go version go1.15 darwin/amd64
```

```
$ sw_vers 
ProductName:	Mac OS X
ProductVersion:	10.13.6
BuildVersion:	17G12034
```

**Expected result**

- `kyma-darwin` doesn't panic
 

**Actual result**

- `kyma-darwin` panics

**Steps to reproduce**

- Run `make build-darwin` using go 1.15 on High Sierra or Mojave
- Run `kyma-darwin`

**Troubleshooting**

- There is no crash is I use go 1.15 on Catalina or go 1.14 on High Sierra/Mojave

**Comment by shorim** [comment]
Hi @bayandin,

It seems that this is an issue with terraform https://github.com/hashicorp/terraform/issues/24287 which was fixed in their latest release `v0.13.0`. Thus, upgrading the terraform version used could resolve this issue. However, we don't have a High Sierra/Mojave machine to test on it. So, could I send you the binary to check if this will actually fix the issue?

**Comment by bayandin** [comment]
> Hi @bayandin,
> 
> It seems that this is an issue with terraform [hashicorp/terraform#24287](https://github.com/hashicorp/terraform/issues/24287) which was fixed in their latest release `v0.13.0`. Thus, upgrading the terraform version used could resolve this issue. However, we don't have a High Sierra/Mojave machine to test on it. So, could I send you the binary to check if this will actually fix the issue?

It would be great if you could share a git branch (or a patch which I could apply) and test it. There're could be a difference during the build stage which I could catch on my older macOS.

**Comment by shorim** [comment]
@bayandin you can use the branch `panicking_on_Sierra_Mojave` from my fork of the cli https://github.com/shorim/cli/tree/panicking_on_Sierra_Mojave.

For your info, updating to terraform v0.13.0 is currently breaking our `provision` command. So, let us see first whether the panic issue will be resolved or not

**Comment by bayandin** [comment]
@shorim yep, that works fine on my Mojave (10.13.6), i.e it doesn't panic straight away. Thanks!

**Comment by shorim** [comment]
@bayandin Great! Thank you for testing it for us.

Just to double check, this panicking issue is not blocking `kyma-cli` release on Homebrew, right? since it will take some time to make the cli fully functioning with the upgraded `terraform` version

**Comment by bayandin** [comment]
No problem!

> Just to double check, this panicking issue is not blocking `kyma-cli` release on Homebrew, right? since it will take some time to make the cli fully functioning with the upgraded `terraform` version

Nope, it's not blocking it; we have switched `kyma-cli` to `go@1.14` until it plays well with `go` 1.15

**Comment by shorim** [comment]
This issue will be addressed in kyma-incubator/hydroform#92

**Comment by clebs** [comment]
Hi @bayandin,
I just did an upgrade of the CLI to k8s 0.18 and terraform 0.12.29: 

Unfortunately, because of conflicts between terraform 0.13 and the k8s go client v0.18 (which we require in Kyma CLI) it is for the time being not possible to upgrade to Terraform 0.13.

I will close the issue, since it has a solution (update Mac OSX) and it may be reopened should the upgrade be possible in a future Terraform release.
